lib/Codegen/Lowering/Distance/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.10)
+
+set(SOURCES
+    DatabaseReader.cpp
+    EditDistance.cpp
+    SentenceDistanceCalculator.cpp
+    WordDistanceCalculator.cpp)
+
+marco_add_library(Distance
+    ${SOURCES}
+
+    LINK_LIBS PUBLIC
+    marco::ast
+    marco::modeling
+    marco::variableFilter
+    ${MLIR_LIBS})
+
+# We use a simple directory structure to store the wordnet database.
+# It will be comprised of some .csv files generated by a python script,
+# which will provide fast access to the wordnet database.
+set(TARGET_DIR "${CMAKE_BINARY_DIR}/wordnet")
+
+# Custom command to create the directory
+add_custom_command(
+    OUTPUT ${TARGET_DIR}
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${TARGET_DIR}
+    COMMENT "Creating directory: ${TARGET_DIR}"
+)
+
+# As opposed to CMAKE_SOURCE_DIR, CMAKE_CURRENT_SOURCE_DIR is
+# the directory of the current CMakeLists.txt file. In this case,
+# it is the directory of this file, which is exactly where the
+# Python script is located.
+set(PYTHON_SCRIPT "${CMAKE_CURRENT_SOURCE_DIR}/install-db.py")
+
+# Custom command to run the Python script
+add_custom_command(
+    OUTPUT "${TARGET_DIR}/.timestamp"  # We use a timestamp file to signify that the script has run.
```

Comment:
".timestamp" is a hidden empty file that signifies that the python "portion" of the process has already been executed, so there's no need to run it again. This is because the python script downloads a relatively large file, so needless repeated execution would be detrimental.

Reply:
Can't the python script just search for previous results?